File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s867221485_commented.py
N = int(input())

s = input()

flag = False
ans = 0
sha = 0
dot = 0
temp = 0
first = True

for i in range(N):
    if s[i] == '#':
        first = False
        if sha <= dot:
            ans += sha
            sha = 0
            dot = 0
            first = True
        else:
            temp += dot
        sha += 1
    else:
        if not first:
            dot += 1

if sha <= dot:
    ans += sha
    dot = 0
else:
    temp += dot

print(ans + dot)  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/C++/s867221485_commented.cpp
#include <iostream>
#include <string>

int main() {
    int N;
    std::cin >> N;

    std::string s;
    std::cin.ignore();
    std::getline(std::cin, s);

    bool flag = false;
    int ans = 0;
    int sha = 0;
    int dot = 0;
    int temp = 0;
    bool first = true;

    for (int i = 0; i < N; i++) {
        if (s[i] == '#') {
            first = false;
            if (sha <= dot) {
                ans += sha;
                sha = 0;
                dot = 0;
                first = true;
            } else {
                temp += dot;
            }
            sha++;
        } else {
            dot++;
        }
    }

    if (sha <= dot) {
        ans += sha;
        dot = 0;
    } else {
        temp += dot;
    }

    std::cout << ans + dot << std::endl;
}

// End of Code
